components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/ApplicationAuthenticatorService.java

```diff
@@ -161,6 +161,18 @@ public LocalAuthenticatorConfig getLocalAuthenticatorByName(String name, String
         return getUserDefinedLocalAuthenticator(name, tenantDomain);
     }
 
+    /**
+     * This returns only SYSTEM defined federated authenticator by name.
+     *
+     * @param name  The name of the federated application authenticator configuration.
+     * @return Retrieved FederatedAuthenticatorConfig.
+     *
+     * @deprecated It is recommended to use getFederatedAuthenticatorByName(String, String) in
+     * org.wso2.carbon.idp.mgt.IdentityProviderManager class, or getSystemFederatedAuthenticatorByName(String)
+     * to retrieve only SYSTEM defined federated application authenticator configuration by name.
+     * which supports retrieving both USER and SYSTEM defined federated application authenticator configuration by name.
```

Comment:
This last sentence is related to the first method referred here not the second one which gives only federated authenticators